app/src/main/java/com/example/instasnap/View/HomePage/HomePageView.java

```diff
@@ -50,19 +50,16 @@ protected void onCreate(Bundle savedInstanceState) {
 
     private void initializeRecyclerViews(){
 
-        ArrayList<Post> posts = new ArrayList<>(); // TODO load posts here
-        ArrayList<Story> stories = new ArrayList<>(); // TODO load stories here
-
         RecyclerView storyRecyclerView = findViewById(R.id.homepage_story_recycler_view);
         RecyclerView postRecyclerView = findViewById(R.id.homepage_post_recycler_view);
 
         _homePageViewModel = new ViewModelProvider(this).get(HomePageViewModel.class);
-        _homePageViewModel.setPostList(posts);
-        _homePageViewModel.setStoryList(stories);
+        _homePageViewModel.setPostList(_allPosts);
+        _homePageViewModel.setStoryList(_stories);
 
         // Create adapter passing in the sample user data
         StoryRecyclerViewAdapter storyRecyclerViewAdapter = new StoryRecyclerViewAdapter();
-        PostRecyclerViewAdapter postRecyclerViewAdapter = new PostRecyclerViewAdapter(posts, _homePageViewModel);
+        PostRecyclerViewAdapter postRecyclerViewAdapter = new PostRecyclerViewAdapter(_users, _homePageViewModel);
```

Comment:
I think its a better idea to pass _allPosts because it is already exists, plus if you pass it here by reference when you change the data of this instance all the other objects holding the reference will change as well (This is a good thing because this class is the viewModel (the logic) therefore all the other views (designs) will update.